script.js

```diff
@@ -0,0 +1,167 @@
+const countriesGrid = document.querySelector('.countries-grid');
+let countryFetchedData = null;
+
+async function fetchCountriesData() {
+    if (countryFetchedData) {
+        return;
+    }
+    try {
+        const response = await fetch('./CountriesData.json');
+        if (!response.ok) {
+            throw `Failed to fetch countries data with status - ${response.status}`;
+        }
+        countryFetchedData = await response.json();
+    } catch (error) {
+        console.error('Error fetching countries data:', error);
+    }
+}
+
+function getDropdownsRegions(data) {
+    const dropdownRegionFilter = document.querySelector('.dropdown-body');
+    const uniqueRegionsSet = new Set(data.map(country => country.region));
+    const uniqueRegions = [...uniqueRegionsSet]
+
+    const dropdownList = document.createElement('ul');
+    const allOption = document.createElement('li');
+    allOption.setAttribute('data-region', 'all');
+    allOption.textContent = 'All';
+    dropdownList.appendChild(allOption);
+
+    uniqueRegions.forEach(region => {
+        const dropdownItem = document.createElement('li');
+        dropdownItem.setAttribute('data-region', region);
+        dropdownItem.textContent = region;
+        dropdownList.appendChild(dropdownItem);
+    });
+    dropdownRegionFilter.appendChild(dropdownList);
+}
+
+function getSearchdata() {
+    const searchInput = document.querySelector('.search-input');
+    searchInput.addEventListener('input', (e) => {
+        searchValue = e.target.value;
+        const filteredData = filterDataSearch(countryFetchedData, searchValue);
+        countriesGrid.innerHTML = '';
+        analyzeMainContainer(filteredData);
+    });
+}
+
```

Comment:
- Never use `innetHTMl`, its a dangerous method

- The name of the function does not reflect what the function is actually doing (adding event listener is not getting data). so please separate your code into helper functions